colexec: fix TestRouterOutputAddBatch when BatchSize=3

asubiotto opened the pull request:
The QuarterSelection test would use BatchSize/4 elements and expect a spill to
disk. With BatchSize=3, the number of elements would be 0, causing the test to
fail since no disk spill occurred.

Release note: None (testing fix).

Fixes #48088

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/48113)
<!-- Reviewable:end -->

yuzefovich reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/48113)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @asubiotto and @yuzefovich)

___
*[pkg/sql/colexec/routers_test.go, line 129 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/48113#-M60U5qH7DUYOAouBIha:-M60U5qH7DUYOAouBIhb:b-ovmrnh) ([raw file](https://github.com/cockroachdb/cockroach/blob/112faaa2f3755ef60fb0077b064cbce3d08c9646/pkg/sql/colexec/routers_test.go#L129)):*
> ```Go
> 			outputBatchSize:  coldata.BatchSize(),
> 			blockedThreshold: getDefaultRouterOutputBlockedThreshold(),
> 			selection:        fullSelection[:len(fullSelection)/3],
> ```

I feel like we should be modifying the test harness, not the test case, to be able to handle such scenarios. Maybe skip the test case if `len(selection) == 0`?


<!-- Sent from Reviewable.io -->

Commit:
colexec: fix TestRouterOutputAddBatch when BatchSize=3

The QuarterSelection test would use BatchSize/4 elements and expect a spill to
disk. With BatchSize=3, the number of elements would be 0, causing the test to
fail since no disk spill occurred.

Release note: None (testing fix).

asubiotto reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/48113)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @yuzefovich)

___
*[pkg/sql/colexec/routers_test.go, line 129 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/48113#-M60U5qH7DUYOAouBIha:-M645Tck6v5H11S-_x2f:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/112faaa2f3755ef60fb0077b064cbce3d08c9646/pkg/sql/colexec/routers_test.go#L129)):*
<details><summary><i>Previously, yuzefovich wrote…</i></summary><blockquote>

I feel like we should be modifying the test harness, not the test case, to be able to handle such scenarios. Maybe skip the test case if `len(selection) == 0`?
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

yuzefovich reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/48113#-:-M65NQbA3RbV6CHOo0Lu:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

asubiotto commented:
bors r=yuzefovich
